### PR TITLE
Dreamer-V2

### PR DESCRIPTION
**Pull request by belerico**
## Summary

This PR introduce the Dreamer-V2 algorithm from https://arxiv.org/abs/2010.02193

## Type of Change

Please select the one relevant option below:

- New feature (non-breaking change that adds functionality)

## Checklist

Please confirm that the following tasks have been completed:

- [x] I have tested my changes locally and they work as expected. (Please describe the tests you performed.)
- [x] I have added unit tests for my changes, or updated existing tests if necessary.
- [x] I have updated the documentation, if applicable.
- [x] I have installed pre-commit and run locally for my code changes.

## Screenshots or Visuals (Optional)

If applicable, please provide screenshots, diagrams, graphs, or videos of the changes, features or the error.

## Additional Information (Optional)

Those are the training loss and rewards con DMC Walker-Walk with the standard hyperparameters found in `dreamer_v2/args.py` 

image
image

Thank you for your contribution! Once you have filled out this template, please ensure that you have assigned the appropriate reviewers and that all tests have passed.

### CODE REVIEW

File: sheeprl/algos/dreamer_v2/utils.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    Returns:
+        The mean and the standard deviation of the distribution of the stochastic state.
+        The sampled stochastic state.
+    """
+    logits = logits.view(*logits.shape[:-1], -1, discrete)
+    dist = Independent(OneHotCategoricalStraightThrough(logits=logits), event_shape)
+    return dist.rsample()
+
+
+def compute_lambda_values(
```
Comment (DavideTr8) [inline sheeprl/algos/dreamer_v2/utils.py:132]: can't this be needed for other algorithms too? maybe to be placed as global utils instead of dreamer's ones?

**Comment by belerico** [comment]
Maybe it's better to account for the action repeat as we have done in Dreamer-V1 @michele-milesi

**Review by michele-milesi — APPROVED** [review]
